src/Blocks/BlockDirt.h

```diff
@@ -96,10 +94,9 @@ class cBlockDirtHandler :
 				continue;
 			}
 
-			BLOCKTYPE AboveDest;
-			NIBBLETYPE AboveMeta;
-			Chunk->GetBlockTypeMeta(BlockX, BlockY + 1, BlockZ, AboveDest, AboveMeta);
-			if (cBlockInfo::GetHandler(AboveDest)->CanDirtGrowGrass(AboveMeta))
```

Comment:
Grass growth is based on light levels, and all CanDirtGrowGrass does is check whether the block is listed as transparent, or if it is a one-hit break type, such as ferns.
So that method was completely wrong in the first place.
By checking the light levels, we are ensuring that we are more Vanilla.